src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java

```diff
@@ -77,6 +77,16 @@
  *   &lt;property name=&quot;classes&quot; value=&quot;java.lang.Boolean&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyTest {
+ *   public void myTest (boolean a, boolean b) {
+ *     Boolean c = new Boolean(a); // violation, instantiation of java.lang.Boolean
+ *                                 // should be avoided
+ *     Boolean d = Boolean.valueOf(b); // OK
+ *   }
+ * }
+ * </pre>
```

Comment:
please add one more example to show usage of `classes` for two types "Integer, Double" as not all user knows how to define few items in array.
https://stackoverflow.com/a/47095501/1015848

for 
https://docs.oracle.com/javase/9/docs/api/java/lang/Double.html
https://docs.oracle.com/javase/9/docs/api/java/lang/Integer.html

Reply:
Thanks for your input @romani 

> looks like it is possible to remove from `tokens` `CLASS_DEF`, please show example of this, I even do not sure what it would mean for validation. Please investigate this.

To clarify:
Do you mean to say I should try to run the check using a configuration like:
```
...
<module name="IllegalInstantiation">
    <property name="classes" value="java.lang.Boolean"/>
    <property name = "tokens" value = "" /> 
</module>
```

Reply:
Thanks for your input @romani 

> looks like it is possible to remove from `tokens` `CLASS_DEF`, please show example of this, I even do not sure what it would mean for validation. Please investigate this.

Could you please elaborate a bit more on this point? I don't completely understand what you mean by >remove from `tokens` `CLASS_DEF`.

Reply:
yes, empty string, and it will work, so users need to know what does it mean and how behavior will change.
Please review logic of Check to understand how it will be different.